src/lj_trace.c

```diff
@@ -736,7 +749,14 @@ void LJ_FASTCALL lj_trace_hot(jit_State *J, const BCIns *pc)
   /* Note: pc is the interpreter bytecode PC here. It's offset by 1. */
   ERRNO_SAVE
   /* Reset hotcount. */
-  hotcount_set(J2GG(J), pc, J->param[JIT_P_hotloop]*HOTCOUNT_LOOP);
+  if (bc_op(pc[-1]) >= BC_FUNCF && bc_op(pc[-1]) <= BC_JFUNCV) {
+    GCproto *pt = (GCproto *)(((char *)(pc-1)) - sizeof(GCproto));
+    lua_assert(pt->hotcount == 0xffff);
```

Comment:
How is this assert true? I don't see where `0xffff` is written anywhere.

Reply:
The hot counter counts down to 0 when it underflows [here in vm_x86.dasc](https://github.com/softdevteam/LuaJIT/pull/11/files#diff-f3a7bbc53d3494d691a380e7bafdd7a5R448) it calls lj_dispatch_call that calls this function.

Reply:
But where is it initialised?

Reply:
It initialised when a function is parsed and its function prototype created in [fs_finish](https://github.com/softdevteam/LuaJIT/pull/11/files#diff-f49c5e8d3541b178bcf63efa1a74216aR1581)

Reply:
Does that mean the assert should be `lua_assert(pt-hotcount == L2J(ls->L)->param[JIT_P_hotfunc])`?

Reply:
No it will always be a 16 bit -1 coming from the interpreter in to this function

Reply:
OK.